paddle/fluid/framework/io/shell.h

```diff
@@ -17,6 +17,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #ifdef _WIN32
+#define NOMINMAX  // msvc max/min macro conflict with std::min/max
```

Comment:
Perhaps is would be good to add this define if it is not already defined:

#ifndef NOMINMAX
#define NOMINMAX
#endif

Reply:
You are right that is a better way. I will add as a separate PR.